web-ng/src/app/components/map/map.component.ts

```diff
@@ -165,11 +175,21 @@ export class MapComponent implements AfterViewInit, OnDestroy {
     });
   }
 
+  private removePolygonOnMap(newFeatureIds: List<string>) {
+    this.map.data.forEach(f => {
+      if (!newFeatureIds.contains(f.getProperty('featureId'))) {
+        this.map.data.remove(f);
+      }
+    });
+  }
+
   private addFeaturesToMap(project: Project, features: List<Feature>) {
     const locationFeatureIds = this.markers.map(m => m.getTitle());
     const geoJsonFeatureIds: String[] = [];
+    const polygonFeatureIds: String[] = [];
     this.map.data.forEach(f => {
       geoJsonFeatureIds.push(f.getProperty('featureId'));
+      polygonFeatureIds.push(f.getProperty('featureId'));
```

Comment:
The `f` here is each geojson object that was imported by `this.map.data.addGeoJson(...)`, it has nothing to do with the polygons.

Reply:
removed.

Reply:
but in marker title is used as identification and in geoJson featureId is used as identification. But in polygon we don't have such thing to setup we only have paths.